src/utils/helpers/clampNumbers.ts

```diff
@@ -0,0 +1,11 @@
+/**
+ * caps the value on a lower and an upper bound
```

Comment:
You should document (and also think about) what the function returns when `value` is NaN. Right now the answer is `NaN`, which is consistent but probably not what we want. Minimum and maximum would both be sensible, but I'm not sure which would be generally better.

Reply:
i think throwing is the right way for NaNs
getting those out of the system is imo even more important than preventing Infinities